### DISCUSSION TITLE
Maestro module not working (item id attribute "data-item-id"?)

### DISCUSSION THREAD

**Discussion by FatherJ-ct** [fvtt-fria-ligan/twilight2000-foundry-vtt#112] [General]
I was trying to get Maestro to work to play sounds for items when rolled.  https://foundryvtt.com/packages/maestro
Maestro Version 0.9.0 says it is compatible with Foundry v10 that I am using and the current t2k core system 2.0.0.

Specifically setting a M4 gun burst sound for the M4 gun item so that when it is rolled it plays the sound. I was unable to get it to work. I tried disabling all other modules (leaving just the t2k core and maestro) and it still did not work. 

The gun sound plays fine from the playlists tab if played manually, the item (m4) has the sound assigned to it, but does not play when a successful hit is rolled. 

I also noticed that the Maestro option to disable the dice roll sound was not working either. The dice roll sound was still happening.

t2k discord user, Daniel@Spelkult.se mentioned using it a YT video https://www.youtube.com/watch?v=z76QQLFu8pg&t=307s  though he didn't show it making the sound on a roll. It is May 15 2022 vid so I believe he is on v9 for foundry on that vid. And I have heard that v10 was pretty big update and broke a lot of modules.

Maestro's configure game settings for the module has an "Item ID attribute" prefilled with "data-item-id" and it says it is used to look for matching item id in chat messages.

Maestro's change log specifically states " GAME SYSTEMS THAT DO NOT INCLUDE AN ITEM ID REFERENCE IN THEIR ITEM ROLL CHAT MESSAGES CANNOT BE USED WITH ITEM TRACKS"

Does the T2k core use the that item id reference?

Thanks and regards,

**Comment by Stefouch** [comment]
No, in T2K, rolling a weapon does not send its ID in a html attribute in the chat message, hence it does not work.

But! This is something I could add.

Currently, the item ID can be found in the roll object in `message.rolls[0].options.itemId`. I don't know if this can help. I need to test Maestro to look how it searches the item's ID.

**Reply by FatherJ-ct** [reply]
I will try that message.rolls[0].options.itemId instead of the data-item-id defaulted in maestro settings when i get a chance. Hopefully that is all it will take. Otherwise i guess it would fall under a feature request to have t2k be able to have maestro working. Definitely a cool immersion add on.

Thank you for the reply and consideration

**Reply by FatherJ-ct** [reply]
I finally got around to trying the message.rolls[0].options.itemId instead of the data-item-id, but it did not work. The item track still plays from the playlists tab fine, but the module seems to have no connection to the dice rolling. It is not playing in reaction to a successful roll and it is also not disabling the dice roll sound itself (which is a configuration item in the module).

**Reply by FatherJ-ct** [reply]
Ah...looks like you beat me to filing an issue over with maestro. I see your https://github.com/death-save/maestro/issues/178

Thank you for following up!
